lib/index.ts

```diff
@@ -148,39 +148,32 @@ export const imagePath = (app: string, file: string) => {
  * @return {string} URL with webroot for a file in an app
  */
 export const generateFilePath = (app: string, type: string, file: string) => {
-	const isCore = window?.OC?.coreApps?.indexOf(app) !== -1
+	const isCore = window?.OC?.coreApps?.includes(app)
+	const isPHP = file.slice(-3) === 'php'
 	let link = getRootUrl()
-	if (file.substring(file.length - 3) === 'php' && !isCore) {
-		link += '/index.php/apps/' + app
+	if (isPHP && !isCore) {
+		link += `/index.php/apps/${app}`
 		if (file !== 'index.php') {
 			link += '/'
 			if (type) {
 				link += encodeURI(type + '/')
 			}
 			link += file
 		}
-	} else if (file.substring(file.length - 3) !== 'php' && !isCore) {
+	} else if (!isPHP && !isCore) {
 		link = getAppRootUrl(app)
 		if (type) {
-			link += '/' + type + '/'
+			link += `/${type}/`
 		}
-		if (link.substring(link.length - 1) !== '/') {
+		if (link.at(-1) !== '/') {
 			link += '/'
 		}
 		link += file
 	} else {
 		if ((app === 'settings' || app === 'core' || app === 'search') && type === 'ajax') {
-			link += '/index.php/'
-		} else {
-			link += '/'
-		}
-		if (!isCore) {
-			link += 'apps/'
-		}
-		if (app !== '') {
-			app += '/'
-			link += app
+			link += '/index.php'
 		}
+		link += `/${app}`
```

Comment:
:one: 

By removing the trailing / on the special ajax handling we can dron the if app empty because we now can simply use a template string with a leading /.

If app is empty only `/` is added if app is not empty `/app` is added.